### DISCUSSION TITLE
Beta test new feature! Fine tuning adjustments: Updated beta

### DISCUSSION THREAD

**Discussion by rsjaffe** [rsjaffe/MIDI2LR#778] [Announcements]
NOTE: First beta had loading error, corrected in this beta version.

Beta testers wanted! Download the attached installer, play around with the new feature, and reply to this message by February 13 with answers to the following questions:
1. Are there bugs in this implementation?
2. Are the options enough? If not, what should they be?
3. What changes in this feature would you like to see and why?
4. Any suggestions as to how to explain this to users? Did you understand my explanation below?
5. Are any adjustments not affected by fineness (other than the ones listed below)?

I've built in a new enhancement to MIDI2LR. Fine adjustments. This allows you to get finer control of the changes in Lightroom when needed. For example, if you're close on exposure but the MIDI controller's changes are too coarse to hit your target exposure exactly, you can engage finer adjustments for your MIDI controller.

Beta testers get to identify issues with this new feature (it is complex, and may cause unexpected issues), and testers can help shape the way the feature is implemented.

There are new commands under the MIDI2LR develop submenu: Fine 2, Fine 4, Fine 8, Fine 16, Fine off, Fine Increase, Fine decrease,  that can be assigned to buttons. These buttons engage a "fineness" amount that affects all your adjustments until you cancel fineness.

For example, pressing *Fine 4* will: restrict the editing range to 1/4 of the usual range, centered, if possible, on the current value for all parameters on your controller. Pressing *Fine 4* when Fine 4 is already in effect will cancel fineness. Pressing any other *Fine number* button when *Fine 4* is active will change the fineness value to the new number—for example *Fine 16* will reduce the editing range to 1/16 of the usual, again, centered on the current value for each parameter. This works the same for any of the *Fine number* commands: the first press engages that fineness, the second press cancels it, pressing any other fineness value switches to that one.

You can also program a *Fine off* button that will cancel fineness no matter what the current value is. Alternatively, you can use the *Fine decrease* and *Fine increase* buttons. Increase doubles fineness (2, 4, 8, 16, etc.) while *Fine decrease* halves it. If you hit *Fine decrease* when fineness is 2, it turns off the fineness adjustment.

Certain adjustments are currently not affected by fineness: *CropMoveHorizontal*, *CropMoveVertical*, quick develop adjustments, *AllSaturationAdjustment*, *PointCurveBlacksUpDown*,  and *PointCurveHighlightsUpDown*. Cropping was a particular trouble point, so testing that is encouraged.

Below are the installers. These install the beta test version and are **not for general distribution**. These versions will have "beta test" on the app window and, once a new release version is available, always prompt you to download the release version (instead of just once).

[beta closed, files removed]

**Comment by rsjaffe** [comment]
For those who downloaded the 6.0.2.1 beta version, please replace with 6.0.2.2 version to avoid Lightroom startup error.

**Comment by 153957** [comment]
I quickly tried the new feature, it seems to work ok.
I have a Midi Fighter Twister, normally using the dials for adjustments, pushing them to reset, and now configured some side buttons to choose the `Fine corrections x`.

When enabling a Fine correction option the dial indeed becomes centered around the current value. However, if I enable Fine corrections, then quickly dial to an extreme value (far from current center), then quickly disable and reenable Fine corrections the dial remains centered on the original/old center, not the new 'extreme value', so as soon as I change the value it picks up from the original center again.

Not certain if this is related to the used Midi controller, Midi2LR, or Lightroom.

Using: Lightroom Classic 13.1, Midi2LR 6.0.2.2, macOS Monterey.

**Reply by rsjaffe** [reply]
That's not expected behavior. Switching off fineness should erase old fineness settings. I'll check next week.

**Comment by 153957** [comment]
Also MIDI2LR crashed when I closed Lightroom, I quickly reopened Lightroom and closed it again to reproduce, but this time MIDI2LR closed without crashing. I can send you the crash report if that would help..

**Reply by rsjaffe** [reply]
Crash report and MIDI2LR.log would help. If you don't know where log file is, open Lightroom and in the help menu under MIDI2LR it'll have an option for diagnostic report, which opens browser at the correct directory.

**Reply by 153957** [reply]
Here is the MIDI2LR log containing 3 section;
- Using 5.4.0.0
- Using 6.0.2.2 with crash at end
- Using 6.0.2.2 without crash at end

I noticed the new log entries are prefixed with a path (to `Misc.h`) containing a user name which does not exist on my machine (I guess its where this was compiled), I've anonymized it in these logs.

[MIDI2LR.log](https://github.com/rsjaffe/MIDI2LR/files/14228637/MIDI2LR.log)
If you'd also like the crash report I can email that to you.
